### PR TITLE
SM Spring Break pull request

### PR DESCRIPTION
**Pull request by regexpressyourself**
# Sam Messina Sprint 2

@reidtevonian @Mikey1848 @taylorflatt I assume you guys are gonna be AFK for most of break. I won't merge this until you all have had a chance to look at it. I'll update this info with any commits I make over break.

## Add Step

- Individual step's Add Step button works like original Add Step Button (completes #14)
  - When a step is added on a current step, the current step is automatically assigned to parent of the new step and the new step is set as a child of the parent

## Edit Step

- Edit step button working (completes #7)
  - Edits reflected immediately (comply with #23)
- Children/parent being properly populated

## Delete Step
- Delete step button is working (completes #8)
- Children/parent relationship removed from list of steps when step is deleted
  - No check yet for max children or anything like that yet

### CODE REVIEW

File: Flowchart/wwwroot/js/React/Components/Canvas.jsx [hunk 1]
```diff
@@ -6,89 +6,331 @@ class Canvas extends React.Component {
...
+                step.title = this.state.titleText;
+                step.description = this.state.descriptionText;
+            }
+            return step;
+        });
+        this.createComponentsFromStepList(newStepList);
+        this.closeEditStepModal();
+    }
+
+    deleteStep(event) {
```
Comment (taylorflatt) [inline Flowchart/wwwroot/js/React/Components/Canvas.jsx:163]: Does this also re-associate the children with new parents? (Graphically and logically)

Comment (taylorflatt) [reply Flowchart/wwwroot/js/React/Components/Canvas.jsx:163]: Ahhh I see how it all works together now. You delete the keys and recreate the graph from the new sets of child/parent relations. Neat.

**Comment by taylorflatt** [comment]
I could be missing it but at what point are you reading the model data from the backend? Or has that not been included yet?

We should probably have a story that is something along the lines of "send data between front and back end" or something. As it stands, passing data back and forth is implied by #31 #24 and #6.

The major objective for back-end this time around is simply saving the data to disk and then retrieving that for modifications. Lemme know what you think and if that could even be incorporated this time around.

**Comment by regexpressyourself** [comment]
@taylorflatt 

Agreed, sending data back and forth is one of my two primary objectives to this sprint. I've been treating #10 as that story for that so far, but I made a new issue for it: #58.

I haven't really touched this problem yet. The main problem is going to be communicating between react and .net. I haven't looked into it much, but I imagine I can just post the flowchart data to a URL and have it get caught by the backend, right?

File: Flowchart/wwwroot/js/React/Components/Canvas.jsx (hunk 1 above)
Comment (regexpressyourself) [reply Flowchart/wwwroot/js/React/Components/Canvas.jsx:163]: Yeah, right now I just nuke everything that has to do with the deleted step. 

Re:
> recreate the graph from the new sets of child/parent relations.

Yeah.....not so much yet. I'm having a tough time figuring out how to handle the rendering stuff properly. Might need to sit down with you and talk it out sometime.

**Comment by regexpressyourself** [comment]
@taylorflatt @reidtevonian @Mikey1848 

Any qualms to what's here? I'm thinking we should merge this in before we get moving on the rest of the sprint.

**Comment by regexpressyourself** [comment]
@taylorflatt @Mikey1848 @reidtevonian 

Hello? Is anybody in there? Just nod if you can hear me.

**Comment by taylorflatt** [comment]
Oh hey. I didn't see the message. I've been holed up working on some labs. But I think it's good to go. Great work man.

**Pull request merged**